README.md
# FasterEnd

Mod for Factorio

An addon for the FasterStart mod who add some mk3 items and a late game recipe for the fusion construction robots.

https://mods.factorio.com/mod/FasterStart

changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.16.5
Date: 31. 08. 2018
  Changes:
    - Few recipes adjustments
    - Update fusion construction robot icon
    - Update few descriptions
  Bugfixes:	
    - Fix recipe bug
---------------------------------------------------------------------------------------------------
Version: 0.16.4
Date: 05. 05. 2018
  Major Features:
    - Added a starter kit with a small modular armor with a small fusion reactor, night vision, exoskeleton, roboport and 25 fusion construction robots.

recipe.lua
data:extend({
	{
    	type = "recipe",
    	name = "fusion-construction-robot",
    	enabled = "false",
    	ingredients =
    		{
      			{"micro-fusion-reactor",1},
      			{"construction-robot",1},
    		},
    	energy_required=15,
    	result = "fusion-construction-robot"
    },

    {
    	type = "recipe",
    	name = "personal-roboport-mk3-equipment",
    	enabled = "false",
    	ingredients =
    	{
      		{"personal-roboport-mk2-equipment",5},
      		{"uranium-fuel-cell",50},
    	},
    	energy_required=30,
    	result = "personal-roboport-mk3-equipment"
    },

    {
    	type = "recipe",
    	name = "key-print",
    	enabled = "false",
    	ingredients =
    	{
      		{"wood",2},
      		{"high-tech-science-pack",1},
      		{"iron-gear-wheel",1}
    	},
    	energy_required=5,
    	result = "key-print"
    },

    {
    	type = "recipe",
    	name = "fusion-logistic-robot",
    	enabled = "false",
    	ingredients =
    	{
    		{"micro-fusion-reactor",1},
      		{"logistic-robot",1},
    	},
    	energy_required=15,
    	result = "fusion-logistic-robot"
    },

    {
    	type = "recipe",
    	name = "micro-fusion-reactor",
    	category = "advanced-crafting",
    	enabled = "false",
    	ingredients =
    	{
    		{"processing-unit",50},
      		{"iron-plate",20},
      		{"copper-cable",10},
      		{"key-print",1}
    	},
    	energy_required = 30,
    	result = "micro-fusion-reactor"
    },

    {
    	type = "recipe",
    	name = "power-armor-mk3",
    	enabled = "false",
    	ingredients =
    	{
     		{"copper-cable",1000},
      		{"processing-unit",800},
      		{"battery-mk3-equipment",10},
      		{"steel-plate",500}
    	},
    	energy_required = 30,
    	result = "power-armor-mk3"
    },

    {
    	type = "recipe",
    	name = "battery-mk3-equipment",
    	enabled = "false",
    	ingredients =
    	{
      		{"uranium-fuel-cell",10},
      		{"battery-mk2-equipment",10}
    	},
    	energy_required = 30,
    	result = "battery-mk3-equipment"
    },
})

prototypes/item.lua
data:extend({
	{
		type = "item",
		name = "fusion-construction-robot",
		icon = "__FasterEnd__/graphics/fusion-construction-robot-icon.png",
		icon_size = 32,
		flags = {"goes-to-main-inventory"},
		subgroup = "logistic-network",
		order = "a[robot]-b[construction-robot]",
		place_result = "fusion-construction-robot",
		stack_size = 50
	},

	{
    	type = "item",
    	name = "personal-roboport-mk3-equipment",
    	icon = "__FasterEnd__/graphics/personal-roboport-mk3-equipment.png",
		icon_size = 32,
   		flags = {"goes-to-main-inventory"},
    	subgroup = "equipment",
    	order = "f",
    	stack_size = 5,
    	placed_as_equipment_result = "personal-roboport-mk3-equipment"
  	},

  	{
    	type = "item",
    	name = "key-print",
    	icon = "__FasterEnd__/graphics/key-print.png",
		icon_size = 32,
    	flags = {"goes-to-main-inventory"},
    	subgroup = "intermediate-product",
    	order = "l",
    	stack_size = 1
  	},

  	{
        type = "item",
        name = "fusion-logistic-robot",
        icon = "__FasterEnd__/graphics/fusion-logistic-robot-icon.png",
		icon_size = 32,
        flags = {"goes-to-main-inventory"},
        subgroup = "logistic-network",
        order = "a[robot]-a[logistic-robot]-b",
        place_result = "fusion-logistic-robot",
        stack_size = 50
    },

    {
        type = "item",
        name = "micro-fusion-reactor",
        icon = "__FasterEnd__/graphics/micro-fusion-reactor.png",
        icon_size = 128,
        flags = {"goes-to-main-inventory"},
        subgroup = "intermediate-product",
        order = "l",
        stack_size = 100,
    },

    {
		type = "item",
		name = "mini-fusion-reactor-equipment",
		icon = "__FasterStart__/graphics/icons/mini-fusion-reactor-equipment.png",
		icon_size = 32,
		placed_as_equipment_result = "mini-fusion-reactor-equipment",
		flags = {"goes-to-main-inventory"},
		subgroup = "equipment",
		order = "a[energy-source]-b[fusion-reactor]",
		stack_size = 1
	},

	{
    	type = "item",
    	name = "battery-mk3-equipment",
    	icon = "__FasterEnd__/graphics/battery-mk3.png",
		icon_size = 32,
    	placed_as_equipment_result = "battery-mk3-equipment",
    	flags = {"goes-to-main-inventory"},
    	subgroup = "equipment",
    	order = "c[battery]-c[battery-equipment-mk3]",
    	stack_size = 50,
    	default_request_amount = 10
  	},
})

prototypes/technology.lua
data:extend({
  {
      type = "technology",
      name = "fusion-construction-robotics",
      icon = "__FasterEnd__/graphics/fusion-construction-robotics.png",
      icon_size = 128,
      effects =
      {
        {
            type = "unlock-recipe",
            recipe = "fusion-construction-robot"
        }
      },
      prerequisites = {"construction-robotics","micro-fusion-reactor"},
      unit =
      {
        count = 100,
        ingredients =
        {
          {"science-pack-1", 1},
          {"science-pack-2", 1},
          {"science-pack-3", 1},
          {"production-science-pack", 1,},
          {"high-tech-science-pack", 1,}
        },
        time = 5
      }
  },
  {
      type = "technology",
      name = "micro-fusion-reactor",
      icon = "__FasterEnd__/graphics/micro-fusion-reactor.png",
      icon_size = 128,
      effects =
      {
                {
            type = "unlock-recipe",
            recipe = "micro-fusion-reactor"
        },
        {
            type = "unlock-recipe",
            recipe = "key-print"
        }
      },
      prerequisites = {"fusion-reactor-equipment"},
      unit =
      {
        count = 100,
        ingredients =
        {
          {"science-pack-1", 1},
          {"science-pack-2", 1},
          {"science-pack-3", 1},
          {"production-science-pack", 1,},
          {"high-tech-science-pack", 1,}
        },
        time = 5
      }
  },
  {
      type = "technology",
      name = "power-armor-3",
      icon = "__FasterEnd__/graphics/power-armor-3.png",
      icon_size = 128,
      effects =
      {
        {
            type = "unlock-recipe",
            recipe = "power-armor-mk3"
        },
        {
            type = "unlock-recipe",
            recipe = "battery-mk3-equipment"
        }
      },
      prerequisites = {"power-armor-2","nuclear-power"},
      unit =
      {
        count = 1000,
        ingredients =
        {
          {"science-pack-1", 1},
          {"science-pack-2", 1},
          {"science-pack-3", 1},
          {"military-science-pack", 1,},
          {"production-science-pack", 1,},
          {"high-tech-science-pack", 1,}
        },
        time = 20
      }
  },
  {
      type = "technology",
      name = "personal-roboport-equipment-3",
      icon = "__base__/graphics/technology/personal-roboport-equipment.png",
      icon_size = 128,
      effects =
      {
        {
            type = "unlock-recipe",
            recipe = "personal-roboport-mk3-equipment"
        }
      },
      prerequisites = {"personal-roboport-equipment-2"},
      unit =
      {
        count = 500,
        ingredients =
        {
          {"science-pack-1", 1},
          {"science-pack-2", 1},
          {"science-pack-3", 1},
          {"military-science-pack", 1,},
          {"production-science-pack", 1,},
          {"high-tech-science-pack", 1,}
        },
        time = 10
      }
  },
  {
      type = "technology",
      name = "fusion-logistic-robotics",
      icon = "__FasterEnd__/graphics/fusion-logistic-robotics.png",
      icon_size = 128,
      effects =
      {
        {
            type = "unlock-recipe",
            recipe = "fusion-logistic-robot"
        }
      },
      prerequisites = {"logistic-robotics","micro-fusion-reactor"},
      unit =
      {
        count = 100,
        ingredients =
        {
          {"science-pack-1", 1},
          {"science-pack-2", 1},
          {"science-pack-3", 1},
          {"production-science-pack", 1,},
          {"high-tech-science-pack", 1,}
        },
        time = 5
      }
  }
})



prototypes/local.lua
data:extend({
	{
        type = "equipment-grid",
    	name = "huge-equipment-grid",
    	width = 20,
    	height = 20,
    	equipment_categories = {"armor"}
  	},
	
	{
        type = "equipment-grid",
		name = "mini-equipment-grid",
		width = 6,
		height = 4,
		equipment_categories = {"armor"}
	},
})

local rob = table.deepcopy(data.raw['roboport-equipment']['personal-roboport-mk2-equipment'])
    rob.name="personal-roboport-mk3-equipment"
    rob.take_result = "personal-roboport-mk3-equipment"
    rob.sprite =
    {
      filename = "__FasterEnd__/graphics/personal-roboport-mk3.png",
      width = 64,
      height = 64,
      priority = "medium"
    }
    rob.energy_source =
    {
      type = "electric",
      buffer_capacity = "1GJ",
      input_flow_limit = "5MW",
      usage_priority = "secondary-input"
    }
    rob.charging_energy = "2MW"
    rob.robot_limit = 50
    rob.construction_radius = 50
    rob.charging_station_count = 10

local bat = table.deepcopy(data.raw['battery-equipment']['battery-mk2-equipment'])
    bat.name = "battery-mk3-equipment"
    bat.sprite =
    {
      filename = "__FasterEnd__/graphics/battery-mk3-equipment.png",
      width = 32,
      height = 64,
      priority = "medium"
    }
    bat.energy_source =
    {
      type = "electric",
      buffer_capacity = "2GJ",
      input_flow_limit = "2GW",
      output_flow_limit = "2GW",
      usage_priority = "terciary"
    }

local huge_armor = table.deepcopy(data.raw['armor']['power-armor-mk2'])
huge_armor.name = "power-armor-mk3"
huge_armor.icon = "__FasterEnd__/graphics/power-armor-mk3.png"
huge_armor.durability = 50000
huge_armor.equipment_grid = "huge-equipment-grid"
huge_armor.inventory_size_bonus = 60
huge_armor.order = "f"
huge_armor.resistances =
    {
      {
        type = "physical",
        decrease = 10,
        percent = 50
      },
      {
        type = "explosion",
        decrease = 30,
        percent = 50
      },
      {
        type = "acid",
        decrease = 8,
        percent = 50
      },
      {
        type = "fire",
        decrease = 5,
        percent = 50
      }
    }

local mini_armor = table.deepcopy(data.raw['armor']['modular-armor'])
mini_armor.name = "mini-power-armor"
mini_armor.icon = "__FasterStart__/graphics/icons/mini-power-armor.png"
mini_armor.resistances = {}
mini_armor.durability = 5000
mini_armor.equipment_grid = "mini-equipment-grid"


local mini_reactor = table.deepcopy(data.raw['generator-equipment']['fusion-reactor-equipment'])
mini_reactor.name = "mini-fusion-reactor-equipment"
mini_reactor.sprite =
    {
      filename = "__FasterStart__/graphics/equipment/mini-fusion-reactor-equipment.png",
      width = 65,
      height = 128,
      priority = "medium"
    }
mini_reactor.shape =
    {
      width = 2,
      height = 4,
      type = "full"
    }
mini_reactor.power = "250kW"

local flr = table.deepcopy(data.raw['logistic-robot']['logistic-robot'])
    flr.name = "fusion-logistic-robot"
    flr.icon = "__FasterEnd__/graphics/fusion-logistic-robot-icon.png"
    flr.energy_per_tick = "0kJ"
    flr.energy_per_move = "0kJ"
    flr.idle =
    {
      filename = "__FasterEnd__/graphics/fusion-logistic-robot.png",
      priority = "high",
      line_length = 16,
      width = 41,
      height = 42,
      frame_count = 1,
      shift = {0.015625, -0.09375},
      direction_count = 16,
      y = 42
    }
    flr.idle_with_cargo =
    {
      filename = "__FasterEnd__/graphics/fusion-logistic-robot.png",
      priority = "high",
      line_length = 16,
      width = 41,
      height = 42,
      frame_count = 1,
      shift = {0.015625, -0.09375},
      direction_count = 16
    }
    flr.in_motion =
    {
      filename = "__FasterEnd__/graphics/fusion-logistic-robot.png",
      priority = "high",
      line_length = 16,
      width = 41,
      height = 42,
      frame_count = 1,
      shift = {0.015625, -0.09375},
      direction_count = 16,
      y = 126
    }
    flr.in_motion_with_cargo =
    {
      filename = "__FasterEnd__/graphics/fusion-logistic-robot.png",
      priority = "high",
      line_length = 16,
      width = 41,
      height = 42,
      frame_count = 1,
      shift = {0.015625, -0.09375},
      direction_count = 16,
      y = 84
    }
    

local fcr = table.deepcopy(data.raw['construction-robot']['construction-robot'])
fcr.name = "fusion-construction-robot"
fcr.icon = "__FasterStart__/graphics/icons/fusion-construction-robot.png"
fcr.minable.result = "fusion-construction-robot"
fcr. energy_per_tick = "0kJ"
fcr.energy_per_move = "0kJ"
fcr.idle.filename = "__FasterStart__/graphics/entity/construction-robot-nuclear.png"
fcr.idle.hr_version.filename = "__FasterStart__/graphics/entity/hr-construction-robot.png"
fcr.in_motion.filename = "__FasterStart__/graphics/entity/construction-robot-nuclear.png"
fcr.in_motion.hr_version.filename = "__FasterStart__/graphics/entity/hr-construction-robot.png"
fcr.shadow_idle.filename = "__FasterStart__/graphics/entity/construction-robot-nuclear-shadow.png"
fcr.shadow_idle.hr_version.filename = "__FasterStart__/graphics/entity/hr-construction-robot-shadow.png"
fcr.shadow_in_motion.filename = "__FasterStart__/graphics/entity/construction-robot-nuclear-shadow.png"
fcr.shadow_in_motion.hr_version.filename = "__FasterStart__/graphics/entity/hr-construction-robot-shadow.png"
fcr.working.filename = "__FasterStart__/graphics/entity/construction-robot-nuclear-working.png"
fcr.working.hr_version.filename = "__FasterStart__/graphics/entity/hr-construction-robot-working.png"
fcr.shadow_working.stripes = 
	util.multiplystripes(2, 
		{
			{
				filename = "__FasterStart__/graphics/entity/construction-robot-nuclear-shadow.png",
				width_in_frames = 16,
				height_in_frames = 1,
			}
		})
		
		
data:extend({mini_armor,mini_reactor,fcr,flr,huge_armor,bat,rob})
